docs/src/tutorials/temporal-properties.md

```diff
@@ -0,0 +1,394 @@
+
+## Auxiliary variables encode evaluations of subformulas along the trace
+
+Next, let us discuss the other auxiliary variables that are introduced by Apalache to check the temporal property.
+These extra variables correspond to parts of the temporal property we want to check.
+These are the following variables with their valuations in state 0:
+
+```
+(* State0 ==
+  RequestWillBeFulfilled_init = FALSE
+    ...
+    /\ ☐(requestedGreen ⇒ ♢isGreen) = FALSE
+    /\ ☐(requestedGreen ⇒ ♢isGreen)_unroll = TRUE
+    /\ requestedGreen ⇒ ♢isGreen = TRUE
+    /\ ♢isGreen = FALSE
+    /\ ♢isGreen_unroll = FALSE
+    ...
+```
+There are three groups of variables: those that look like formulas, e.g. `☐(requestedGreen ⇒ ♢isGreen)`,
+those that look like formulas and end with `_unroll`,
+e.g., and the variable `RequestWillBeFulfilled_init`.
+
+Let's focus on the non-`_unroll` variables that look like formulas
+first.
+
+Recall that the temporal property we want to check is `[](requestedGreen => <>isGreen)`.
+That's also the name of one of the variables: The value of the variable 
+`☐(requestedGreen ⇒ ♢isGreen)` tells us whether starting in the current state, the
+formula `[](requestedGreen => <>isGreen)` holds. Since we are looking at a counterexample, it is not
+surprising that the formula does not hold in state 0.
+
+Similarly, the variable `requestedGreen ⇒ ♢isGreen` tells us whether
+the property `requestedGreen ⇒ ♢isGreen` holds at the current state.
+It might be surprising to see that the property holds
+but recall that in state 0, `requestedGreen = FALSE`, so the implication is satisfied.
+Finally, we have the variable `♢isGreen`, which is false, telling
+us that along the execution, `isGreen` will never be true.
+
+You might already have noticed the pattern of which formulas appear as variables.
+Take our property `[](requestedGreen => <>isGreen)`.
+The syntax tree of this formula looks like this:
+
+![The syntax tree for the formula [](requestedGreen => <>isGreen)](img/syntaxtree.png)
+
+For each node of the syntax tree where the formula contains a temporal operator,
+there is an auxiliary variable. And as mentioned before, the value of 
+the variable in a state tells us whether from that state, the subformula is true. In this particular example, the formulas that correspond to variables in the encoding are filled with orange in the syntax tree.
+
+What about the `_unroll` variables? There is one `_unroll` variable for each temporal operator in the formula.
+For example, `☐(requestedGreen ⇒ ♢isGreen)_unroll` is the unroll-variable for the
+leading box operator. These are necessary because to know whether a formula like
+`[]isGreen` holds in the last state of the loop, we need to know whether
+`isGreen` holds in all states of the loop. However, when we are in the last state of the loop,
+we won't see the state of the loop again! So we need to store this information when we traverse the loop.
```

Comment:
I don't understand what state "we won't see the state of the loop again".

Also, if something is done by an algorithm, I prefer to say it that we instead of using a generic "we".

Reply:
Rewrote this a bit and replaced "we" where appropriate